pyiceberg/schema.py

```diff
@@ -221,6 +227,11 @@ def find_type(self, name_or_id: Union[str, int], case_sensitive: bool = True) ->
     def highest_field_id(self) -> int:
         return max(self._lazy_id_to_name.keys(), default=0)
 
+    def name_mapping(self) -> NameMapping:
```

Comment:
Could this be a `@cached_property` since schema should be immutable?

Reply:
That's a great suggestion 👍